services/continuous_querier/service.go

```diff
@@ -82,20 +85,26 @@ type Service struct {
 	// lastRuns maps CQ name to last time it was run.
 	mu       sync.RWMutex
 	lastRuns map[string]time.Time
+	cqLocks  map[string]*sync.RWMutex
```

Comment:
Looking below, this lock would likely be better implemented as a channel instead. Something similar to this pattern:

```
lock := make(chan struct{}, 1)
lock <- struct{}{}

// acquire the lock
select {
case <-lock:
    // acquired lock
    defer func() {
        lock <- struct{}{}
    }()
default:
    // unable to acquire the lock
    return
}

// you now have the lock
...
```